### DISCUSSION TITLE
[How To Question]:  Open random existing note

### DISCUSSION THREAD

**Discussion by dzintars** [zk-org/zk-nvim#94] [General]
How would some retrieve random existing note?

All my notes have flat file structure:
![image](https://user-images.githubusercontent.com/547420/208951465-1296b89c-62da-4668-9aa6-7e4cc88f5380.png)
What i would like to have is a function/command to open random note to revisit/refactor it.
I also have notes in other directories. For example `./Daily/*` with different file names if that is important.

What I imagine is a way to return note at random index of the total number of all `ZK` indexed notes.

**Comment by dzintars** [comment]
Found this: `zk list --quiet --format full --sort random --limit 1`

**Reply by mickael-menu** [reply]
That's right, `--sort random` is the key here. You can use it with `zk-nvim` too: 

```
:ZkNotes { sort = {"random"}, limit = 1 }
```

**Comment by dzintars** [comment]
```lua
    vim.keymap.set(
      'n',
      '<leader>zu',
      '<cmd>ZkNotes { sort = {"random"}, limit = 1 }<cr>',
      { desc = 'zk-nvim | Open random note' }
    )
```
